packages/flutter/lib/src/widgets/animated_child_switcher.dart

```diff
@@ -125,31 +205,36 @@ class _AnimatedChildSwitcherState extends State<AnimatedChildSwitcher> with Tick
     super.dispose();
   }
 
+  bool get hasNewChild => widget.child != null;
+  bool get hasOldChild => _currentChild?.widget != null;
+
   @override
-  Widget build(BuildContext context) {
-    if (widget.child != _currentChild.widget) {
-      addEntry(true);
+  void didUpdateWidget(AnimatedChildSwitcher oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (hasNewChild != hasOldChild || hasNewChild && !Widget.canUpdate(widget.child, _currentChild.widget)) {
+      addEntry(animate: true);
     }
-    final List<Widget> children = <Widget>[];
```

Comment:
you need to update _currentChild.widget to the new widget.child if you don't add an entry.

Reply:
Wait, why? addEntry isn't called if there's not a new widget.child.  Maybe I just misunderstand this comment...